### PR TITLE
Enables autocomplete for research notes

### LINKED ISSUE #8223: Setup Atwho data module for autocompletion
**Please describe the desired behavior.**
Describe the solution you'd like.

Following the discussion [here ](https://github.com/publiclab/PublicLab.Editor/issues/8#issuecomment-665628384) we'll need to setup Atwho.js data module to able to autocomplete authors, wiki pages, and emojis.
I'm going to be making a PR for this soon

**Additional context (optional)**
Feel free to include any applicable: code, screenshots, or gifs.

### PR DESCRIPTION
**Pull request by Shulammite-Aso**
Fixes #8223 
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by Shulammite-Aso** [comment]
Hey @jywarren can you check if this works? Hoping you've seen my comment here https://github.com/publiclab/plots2/issues/8223#issuecomment-668792746

@cesswairimu @sagarpreet-chadha maybe you'll be of help too.

Thanks!

**Comment by Shulammite-Aso** [comment]
Didn't commit package.json and lock file changes after installing At.js, since the autocompletion was already working for comments without At.js.

**Comment by jywarren** [comment]
Hi, this looks all right... but i don't see it working in GitPod at least -- does it work for you? What test strings do you use? 

image

Maybe we should add a simple system test (it'll be like a Jest test) to confirm this works?

**Comment by jywarren** [comment]
A simple system test could be to add a `@` or equivalent test string in a test very much like this one: 

https://github.com/publiclab/plots2/blob/4d005ccb7d34b6c674e2962519f17183644598c0/test/system/post_test.rb#L18-L32

If we end the body content string with a trigger character like `@` or `#`, we should be able to assert that the autocomplete suggestions list appears by referencing its CSS class.

**Comment by Shulammite-Aso** [comment]
@jywarren , i knew this wasn't working, at least from my end. I only thought perhaps i wasn't doing something right or i may be having some restrictions,  so i just pushed it for somebody else to confirm if it works or not.
 
But i've just figured out how to go about it which i'll be pushing soon.  We just need to move the Atwho.PLE.js file from the example folder here https://github.com/publiclab/PublicLab.Editor/tree/main/examples/data to the src folder. I'll just open a PR at editor repo, then this feature will be available downstream after the next release.

**Comment by Shulammite-Aso** [comment]
Will won't need to make any changes here in plots2 for `@` and `#`  watchers to work here,  but I'm still  trying to figure out if we'll need to make some changes, like have the emoji.js file here plots2 for the emoji watcher to work, because I don't see it working in plots2 yet.

**Comment by sagarpreet-chadha** [comment]
Hey @Shulammite-Aso , correct we will not change anything in `plots2`, only we will bump the version number 👍 
But saying that I am not sure about how emojis are bundled by `gruntjs` in PL.editor so we may need to copy emojis file here in plots2.

**Comment by Shulammite-Aso** [comment]
> But saying that I am not sure about how emojis are bundled by gruntjs in PL.editor so we may need to copy emojis file here in plots2

Have copied this to `assets` in plots2 but haven't seen it working. Also linked the file from the `rich.html.erb` file and still didn't work. 

You have any idea how else to go about this?

File: app/views/editor/rich.html.erb [hunk 1]
```diff
@@ -283,6 +288,9 @@
         <a href="https://publiclab.org/post?n=23853" class="bug-btn" target="_blank">Found an issue with the editor? Report a bug.</a>
   </div>
 
+  <script src="/emoji.js" type="text/javascript"></script>
```
Comment (sagarpreet-chadha) [inline app/views/editor/rich.html.erb:291]: Do we have this file?

File: app/views/editor/rich.html.erb [hunk 2]
```diff
@@ -18,6 +18,11 @@
 <%= javascript_include_tag('/lib/leaflet-spin/example/spin/dist/spin.min.js') %>
 <%= javascript_include_tag('/lib/leaflet-spin/example/leaflet.spin.min.js') %>
 
+<link
+      href="/lib/at.js/dist/css/jquery.atwho.min.css"
+      rel="stylesheet"/>
+    <script src="/lib/at.js/dist/js/jquery.atwho.min.js"></script>
```
Comment (sagarpreet-chadha) [inline app/views/editor/rich.html.erb:24]: Do we have `atwho` in package.json?

**Comment by sagarpreet-chadha** [comment]
So only emojis are not working? Can you commit the new changes to this PR?

**Comment by Shulammite-Aso** [comment]
> So only emojis are not working? Can you commit the new changes to this PR?

@sagarpreet-chadha   there's no new changes to this. We already made the needed changes at the Editor repository here  https://github.com/publiclab/PublicLab.Editor/pull/606 

We can either close this or if we figure out how to make Emojis here, then we can do it in this PR

**Comment by cesswairimu** [comment]
Closing this based on @Shulammite-Aso comment above :point_up: Feel free to reopen if I missed smth. Thanks all

**Pull request closed without merging**